source/slang/source-loc.h

```diff
@@ -82,17 +113,21 @@ class SourceFile : public RefObject
     /// The actual contents of the file.
     UnownedStringSlice content;
 
-    // The range of source locations that the span covers
-    SourceRange sourceRange;
-
+    protected:
     // In order to speed up lookup of line number information,
     // we will cache the starting offset of each line break in
     // the input file:
-    List<UInt> lineBreakOffsets;
+    List<uint32_t> m_lineBreakOffsets;
 };
 
 struct SourceManager;
 
+enum class SourceLocType
+{
+    Normal,                 ///< Takes into account #line directives
+    Original,               ///< Ignores #line directives - humane location as seen in the actual file
```

Comment:
We need to establish the terminology here. The whole "spelling location" thing is something derived from clang, which has very carefully defined terms here. "Normal" and "original" are maybe too generic in this case, so we should probably either keep using Clang terminology, or invent our own terms of art that can be used without confusion.

Reply:
Ok I've changed Original to Spelling. I didn't like the 'Spelling' name when I refactored the original code - it took me quite a while to figure out what it meant from the source. I'm not going to look its meaning up on clang compiler. But a comment on the enum will hopefully mean that's now not so bad.

Reply:
Quickly glancing at some clang docs, I think I've flubbed their terminology anyway. In the context of locations, the "spelling" thing works more like:

- A "spelling location" refers to the location where a token was actually written, like the `bar` in `#define FOO bar`

- An "expansion location" refers to the location where a token appeared in the macro-expanded code, so that in `doIt(FOO)` we might have an identifier token for `bar` with an expansion location that points to the `FOO`. It is always possible to map from an expansion location back to the spelling location (but not vice versa)

We don't currently allocate additional ranges for macro expansion (just token pastes) so this distinction doesn't apply in Slang today. What we are talking about right now is the mapping proces from an integer source location to a "humane" file/line/column representation, which can either take `#line` directives into account, or not.

Given that, I might advocate that the terms we should use are "nominal" and "actual":

- A "nominal" location would be one that takes `#line` into account, and is thus what we should report in diagnostics by default

- An "actual" location would be one that doesn't take `#line` into account, and is thus what we want when we really want to answer the question like "what is the path to the source file doing this `#include` right now?"

The spelling/expansion and nominal/actual axes are independent, in that I can ask for the actual file/line of an expansion location, the nominal file/line of a spelling location, etc.

Thoughts? I'm just throwing terminology at the wall at this point.

Reply:
Actual is pretty good. Perhaps Normal is better than Nominal - it's more commonly used and understood word. 
Just to explain my 'I'm not going to look its meaning up on clang compiler.' comment cos it could be taken inappropriately - I just meant if I don't understand a term in the slang code base, my default course of action isn't to look at clang or any other compiler. If it's a very common term used across compilers then fair game. If not, well not so much. Either way a comment explaining helps a lot.